### PR TITLE
lifetime: circular dependency; Fix #77

### LINKED ISSUE #77: lifetime: circular dependency
Hi
I find a import problems about /opt/rucio/lib/rucio/tests/test_lifetime.py
This test will send request to trigger lib/rucio/web/rest/lifetime_exception.py
but it seems a circular import occurs in this .py ?
 
```
(.venv) [root@rucio-test03 rucio]# python -i lib/rucio/web/rest/lifetime_exception.py
Traceback (most recent call last):
  File "lib/rucio/web/rest/lifetime_exception.py", line 17, in <module>
    from rucio.api.lifetime_exception import list_exceptions, add_exception, update_exception
  File "/opt/rucio/lib/rucio/api/lifetime_exception.py", line 14, in <module>
    from rucio.core import lifetime_exception
  File "/opt/rucio/lib/rucio/core/lifetime_exception.py", line 22, in <module>
    import rucio.common.policy
  File "/opt/rucio/lib/rucio/common/policy.py", line 23, in <module>
    import rucio.core.did
  File "/opt/rucio/lib/rucio/core/did.py", line 35, in <module>
    import rucio.core.rule
  File "/opt/rucio/lib/rucio/core/rule.py", line 32, in <module>
    import rucio.core.lock  # import get_replica_locks, get_files_and_replica_locks_of_dataset
  File "/opt/rucio/lib/rucio/core/lock.py", line 27, in <module>
    from rucio.core.lifetime_exception import define_eol
ImportError: cannot import name define_eol
```
 
In python shell:
 
```
>>> import rucio.core.lifetime_exception
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/rucio/lib/rucio/core/lifetime_exception.py", line 22, in <module>
    import rucio.common.policy
  File "/opt/rucio/lib/rucio/common/policy.py", line 23, in <module>
    import rucio.core.did
  File "/opt/rucio/lib/rucio/core/did.py", line 35, in <module>
    import rucio.core.rule
  File "/opt/rucio/lib/rucio/core/rule.py", line 32, in <module>
    import rucio.core.lock  # import get_replica_locks, get_files_and_replica_locks_of_dataset
  File "/opt/rucio/lib/rucio/core/lock.py", line 27, in <module>
    from rucio.core.lifetime_exception import define_eol
ImportError: cannot import name define_eol
```
 
Best RegardEdward Wu
ASGC

_Migrated from JIRA [RUCIO-2835](https://its.cern.ch/jira/browse/RUCIO-2835)_

### PR DESCRIPTION
**Pull request by cserf**
lifetime: circular dependency; Fix #77

### CODE REVIEW

File: lib/rucio/common/policy.py [hunk 1]
```diff
@@ -16,20 +16,13 @@
...
 from dogpile.cache import make_region
 from dogpile.cache.api import NoValue
 
-import rucio.core.did
-import rucio.core.lock
-import rucio.core.rule
-
-from rucio.db.sqla.constants import DIDType
-from rucio.db.sqla.session import read_session, transactional_session
 from rucio.common.config import config_get
```
Comment (vingar) [inline lib/rucio/common/policy.py:25]: sort import

File: lib/rucio/common/policy.py [hunk 2]
```diff
@@ -79,92 +72,17 @@ def get_lifetime_policy():
...
-    :param rses:     List of RSEs.
-    :param lifetime: Lifetime.
-    :param session:  The database session in use.
-    """
-
+def policy_filter(function):
+    mapping = {'atlas': ['get_scratch_policy', 'archive_localgroupdisk_datasets']}
     vo_name = get_vo()
-    scratchdisk_lifetime = get_scratchdisk_lifetime()
-    if vo_name == 'atlas':
```
Comment (vingar) [inline lib/rucio/common/policy.py]: if policy ==

File: lib/rucio/common/policy.py [hunk 3]
```diff
@@ -79,92 +72,17 @@ def get_lifetime_policy():
...
-    vo_name = get_vo()
-    if vo_name != 'atlas':
-        return
-
-    rses_to_rebalance = []
-
-    # Check if the archival dataset already exists
-    try:
-        rucio.core.did.get_did(scope='archive', name=name, session=session)
+    if vo_name in mapping and function.__name__ in mapping[vo_name]:
```
Comment (vingar) [inline lib/rucio/common/policy.py]: policy instead of vo

File: lib/rucio/common/policy.py (hunk 1 above)
Comment (cserf) [reply lib/rucio/common/policy.py:25]: ??? That's sorted. First system import, then other packages

Comment (vingar) [reply lib/rucio/common/policy.py:25]: hmmm 🔢 
from rucio.common.config import config_get
from rucio.db.sqla.constants import DIDType
from rucio.db.sqla.session import read_session, transactional_session

I think I missed the - removed lines

**Review by bari12 — APPROVED** [review]

**Pull request merged**